validators/validation.js

```diff
@@ -0,0 +1,15 @@
+const Joi = require("joi");
+const pattern = /[`!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?~][0-9]/;
+exports.signUpSchema = Joi.object({
+  email: Joi.string().email().lowercase().required(),
+  password: Joi.string().pattern(new RegExp(pattern)).min(3).max(15).required(),
+  role: Joi.any().allow("admin", "student").required(),
+  firstname: Joi.string().min(2).required(),
+  lastname: Joi.string().min(2).required(),
+  gender: Joi.any().allow("male", "female", "M", "f").required(),
```

Comment:
It's not a good idea to take two different values for a single field. e.g. `male` and `m` for Male is not good enough. 
You are in control of the backend and you should not cause confusion for yourself or any other person. Stick to F and M or Male and Female.
Treat it like Enum